include/swift/AST/DiagnosticsSema.def

```diff
@@ -273,7 +273,7 @@ ERROR(cannot_infer_closure_parameter_type,none,
 ERROR(cannot_infer_closure_type,none,
       "unable to infer closure type in the current context", ())
 ERROR(cannot_infer_closure_result_type,none,
-      "unable to infer%select{ complex|}0 closure return type; "
+      "unable to infer multiple-statement closure return type; "
       "add explicit type to disambiguate", (bool))
```

Comment:
Based on some discussion with other folks, it looks like the single-expression case (when the boolean is true) cannot be hit at all. So you could remove the `(bool)` from the below line and update the place in `CSDiagnostics.cpp` emitting the diagnostic accordingly.

Wording-wise, perhaps it would be better to rephrase this to:

```suggestion
      "cannot infer return type for closure with multiple statements; "
      "add explicit type to disambiguate", ())
```

Reply:
Yes, I think that single-statement cloure check might just be an artifact from the CSDiag days which is no longer useful. Also I like the phrasing @varungandhi-apple suggested.